contracts/MultisigDAOUpgradeable.sol

```diff
@@ -0,0 +1,706 @@
+        uint32 _lockDuration,
+        uint96 _proposalThreshold,
+        uint8 _quorumDivisor
+    ) public initializer {
+        maxProposalTargets = _maxProposalTargets;
+        govToken = IERC20(_govToken);
+        warmUp = _warmUp;
+        coolingOff = _coolingOff;
+        proposalDuration = _proposalDuration;
+        lockDuration = _lockDuration;
+        proposalThreshold = _proposalThreshold;
+        quorumDivisor = _quorumDivisor;
+    }
+
+    function initializeMultisig(address _multisig) external {
+        require(!multisigInitialized, "DAO: Multisig address already initialized");
+        multisigInitialized = true;
+        multisig = _multisig;
+        emit SetMultisig(multisig);
+    }
+
+    function name() external pure returns (string memory) {
+        return "MultisigDAOUpgradeable";
+    }
+
+    /**
+     * @notice Payable fallback to accept ETH payments.
+     */
+    receive() external payable {}
+
+    /**
+     * @notice Gets the total amount staked by a given user.
+     * @param account The user account address.
+     */
+    function getStaked(address account) public view returns (uint96) {
+        return stakers[account].stakedAmount;
+    }
+
+    /**
+     * @notice Stakes governance tokens allowing users voting rights.
+     * @dev requires users to have approved this contract to transfer their
+     *      governance tokens.
+     * @param amount the amount of governance tokens to stake.
+     */
+    function stake(uint96 amount) external {
+        govToken.safeTransferFrom(msg.sender, address(this), amount);
+        stakers[msg.sender].stakedAmount = stakers[msg.sender]
+            .stakedAmount
+            .add96(amount);
+        totalStaked = totalStaked.add(amount);
+        emit UserStake(msg.sender, amount);
+    }
+
+    /**
+     * @notice Withdraws from a given users stake.
+     * @param amount the amount of governance tokens to withdraw.
+     */
+    function withdraw(uint96 amount) external {
+        Stake memory staker = stakers[msg.sender];
+        require(
+            staker.lockedUntil < block.timestamp,
+            "DAO: Tokens are vote locked"
+        );
+        // SafeMath sub will throw if amount > staked
+        stakers[msg.sender].stakedAmount = staker.stakedAmount.sub96(amount);
+        totalStaked = totalStaked.sub(amount);
+        govToken.safeTransfer(msg.sender, amount);
+        emit UserWithdraw(msg.sender, amount);
+    }
+
+
+    /**
+     * @notice Proposes a function execution on a contract by the governance contract.
+     * @param targets the target contracts to execute the proposalData on.
+     * @param  proposalData ABI encoded data containing the function signature and parameters to be
+     *         executed as part of this proposal.
+     */
+    function propose(address[] memory targets, bytes[] memory proposalData, bool _allowMultisig)
+        public
+        onlyMultisigOrStaker()
+    {
+        uint96 userStaked = getStaked(msg.sender);
```

Comment:
Checking this is intended behavior:

For the multisig to propose, they must have more than the minimum required amount staked

Reply:
Ah good catch. That's not intended.

I've also changed the tests to use a non-staked account as the multisig.